sdk/python/feast/feature_store.py

```diff
@@ -480,7 +535,10 @@ def tqdm_builder(length):
 
     @log_exceptions_and_usage
     def get_online_features(
-        self, feature_refs: List[str], entity_rows: List[Dict[str, Any]],
+        self,
+        features: Union[List[str], FeatureService],
```

Comment:
One thing that stands out here is that we are passing in an object vs a string and we are not pulling the latest version of the feature service from the registry. The risk with the current approach is that the user can pass in an outdated object, which could result in missing features being queried. What do you think about either using a string or pulling the latest feature service from the registry. The registry does have caching, so it shouldnt be a problem I think.

Reply:
In what circumstances could users pass in an outdated object? Say by getting a feature view from the feature store, making making changes to the feature view and applying, but passing in the old references to the feature service? 

Getting the latest from the registry is reasonable I think, yes. I'll update

Reply:
Yea, the objects are mutable. They can make changes and use the changed object.